### PR TITLE
[ML] Label anomalies with  multi_bucket_impact

### PR DESCRIPTION
**Pull request by edsavage**
Add the multi_bucket_impact field to record results.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyRecord.java [hunk 1]
```diff
@@ -198,6 +202,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(jobId);
         out.writeInt(detectorIndex);
         out.writeDouble(probability);
+        out.writeDouble(impact);
```
Comment (droberts195) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyRecord.java]: To make this backwards compatible it needs to be:

```
    if (out.version().onOrAfter(Version.V_6_5_0)) {
        out.writeOptionalDouble(impact);
    }
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyRecord.java [hunk 2]
```diff
@@ -247,6 +252,7 @@ XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws I
         builder.field(Job.ID.getPreferredName(), jobId);
         builder.field(Result.RESULT_TYPE.getPreferredName(), RESULT_TYPE_VALUE);
         builder.field(PROBABILITY.getPreferredName(), probability);
+        builder.field(IMPACT.getPreferredName(), impact);
```
Comment (droberts195) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyRecord.java]: When the type is changed to `Double` this should be wrapped in a `null` check so we write nothing if it's `null`.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ml/job/results/AnomalyRecordTests.java [hunk 3]
```diff
@@ -38,6 +38,7 @@ public static AnomalyRecord createTestInstance(String jobId) {
         anomalyRecord.setActual(Collections.singletonList(randomDouble()));
         anomalyRecord.setTypical(Collections.singletonList(randomDouble()));
         anomalyRecord.setProbability(randomDouble());
+        anomalyRecord.setImpact(randomDouble());
```
Comment (droberts195) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/ml/job/results/AnomalyRecordTests.java]: To confirm `null` works once the type is changed to `Double`, surround this line with `if (randomBoolean()) {`.

Same for the server-side file.

**Review by droberts195 — COMMENTED** [review]
I suggested a renaming change but I think IntelliJ's refactoring functionality (right click -> refactor -> rename) should make light work of it.  It has the option to automatically rename arguments and accessors.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ml/job/results/AnomalyRecordTests.java (hunk 3 above)
Comment (edsavage) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/ml/job/results/AnomalyRecordTests.java]: Will do - thanks for the quick review @droberts195

**Review by droberts195 — APPROVED** [review]
LGTM as long as the CI goes green.

If there's a failure it's most likely related to the BWC aspects.

**Pull request merged**